tools/LogCollector/source/update_mgmt_health_check.py

```diff
@@ -1,268 +1,392 @@
 
-    return "".join(agent_status)
+def check_agent_service_endpoint():
+    rule_id = "Linux.AgentServiceConnectivityCheck"
+    rule_group_id = "connectivity"
 
-def check_hybrid_worker_running():
-    return grep_for_process("worker", 
-                            ["python /opt/microsoft/omsconfig/modules/nxOMSAutomationWorker/DSCResources/MSFT_nxOMSAutomationWorkerResource/automationworker/worker/hybridworker.py"], 
-                            "Hybrid worker")
-
-def check_hybrid_worker_installed():
-    workspace = get_workspace()
+    agent_endpoint = get_agent_endpoint()
+    if  agent_endpoint is None:
+        write_log_output(rule_id, rule_group_id, status_failed, "UnableToGetEndpoint", "Unable to get the registration (agent service) endpoint")
+    elif  check_endpoint(None, agent_endpoint):
+        write_log_output(rule_id, rule_group_id, status_passed, empty_failure_reason, "TCP test for {" + agent_endpoint + "} (port 443) succeeded", agent_endpoint)
+    else:
+        write_log_output(rule_id, rule_group_id, status_failed, empty_failure_reason, "TCP test for {" + agent_endpoint + "} (port 443) failed", agent_endpoint)
 
-    if workspace is not None:
-        worker_conf_path = "/var/opt/microsoft/omsagent/" + workspace + "/state/automationworker/worker.conf"
+def check_jrds_endpoint(workspace):
+    rule_id = "Linux.JRDSConnectivityCheck"
+    rule_group_id = "connectivity"
 
-        if os.path.isfile(worker_conf_path):
-            return "worker.conf exists. Hybrid worker looks to be installed. \n"
-        else:
-            return "worker.conf does not exist. Is hybrid worker installed? \n"
+    jrds_endpoint = get_jrds_endpoint(workspace)
+    if jrds_endpoint is None:
+        write_log_output(rule_id, rule_group_id, status_failed, "UnableToGetEndpoint", "Unable to get the operations (JRDS) endpoint")
+    elif jrds_endpoint is not None and check_endpoint(workspace, jrds_endpoint):
+        write_log_output(rule_id, rule_group_id, status_passed, empty_failure_reason, "TCP test for {" + jrds_endpoint + "} (port 443) succeeded", jrds_endpoint)
     else:
-        return "Checking hybrid worker installed...workspace not found. Is OMS Agent installed?"
+        write_log_output(rule_id, rule_group_id, status_failed, empty_failure_reason, "TCP test for {" + jrds_endpoint + "} (port 443) failed", jrds_endpoint)
 
-def check_oms_agent_running():
-    return grep_for_process("omsagent", ["omsagent.log", "omsagent.conf"], "OMS Agent")
+def check_log_analytics_endpoints():
+    rule_id = "Linux.LogAnalyticsConnectivityCheck"
+    rule_group_id = "connectivity"
 
-def grep_for_process(process_name, search_criteria, output_name):
-    grep_output = os.popen("ps aux | grep " + process_name).read()
+    i = 0
+    if is_fairfax_region() is True:
+        fairfax_log_analytics_endpoints = ["usge-jobruntimedata-prod-1.usgovtrafficmanager.net", "usge-agentservice-prod-1.usgovtrafficmanager.net",
```

Comment:
please consider refactoring with endpoints as parameter

Reply:
Ignoring this comment.
Since these endpoints are not going to change it be easy for customers to use.